lib/wibox/layout/manual.lua

```diff
@@ -0,0 +1,211 @@
+                }
+            })
+            -- Trick to ensure compatibility with `awful.placement`
+            gtable.crush(pt, geo._new_geo or {})
+        end
+
+        assert(pt.x)
+        assert(pt.y)
+        assert(pt.width or w)
+        assert(pt.height or h)
+
+        table.insert(res, base.place_widget_at(
+            v, pt.x, pt.y, pt.width or w, pt.height or h
+        ))
+    end
+
+    return res
+end
+
+function manual_layout:add(...)
+    local wdgs = {...}
+    local old_count = #self._private.widgets
+    gtable.merge(self._private.widgets, {...})
+
+    -- Add the points
+    for k, v in ipairs(wdgs) do
+        if v.point then
+            self._private.pos[old_count+k] = v.point
+        end
+    end
+
+    self:emit_signal("widget::layout_changed")
+end
+
+--- Add a widget at a specific point.
+--
+-- The point can either be a function or a table. The table follow the generic
+-- geometry format used elsewhere in Awesome.
+--
+-- * *x*: The horizontal position.
+-- * *y*: The vertical position.
+-- * *width*: The width.
+-- * *height*: The height.
+--
+-- If a function is used, it follows the same prototype as `awful.placement`
+-- functions.
+--
+-- * *geo*:
+--     * *x*: The horizontal position (always 0).
+--     * *y*: The vertical position (always 0).
+--     * *width*: The width.
+--     * *height*: The height.
+--     * *geometry*: A function to get or set the geometry (for compatibility)
+-- * *args*:
+--     * *parent* The layout own geometry
+--         * *x*: The horizontal position (always 0).
+--         * *y*: The vertical position (always 0).
+--         * *width*: The width.
+--         * *height*: The height.
+--         * *geometry*: A function to get or set the geometry (for compatibility)
+--
+--@DOC_wibox_layout_manual_add_at_EXAMPLE@
+-- @tparam widget widget The widget.
+-- @tparam table|function point Either an `{x=x,y=y}` table or a function
+--  returning the new geometry.
+function manual_layout:add_at(widget, point)
+    self._private.pos[#self._private.widgets+1] = point
+    self:add(widget)
+end
+
+--- Move a widget (by index).
+-- @tparam number index The widget index.
+-- @tparam table|function point A new point value.
+-- @see add_at
+function manual_layout:move(index, point)
+    assert(self._private.pos[index])
+    self._private.pos[index] = point
+    self:emit_signal( "widget::layout_changed" )
+end
+
+--- Move a widget.
+--
+--@DOC_wibox_layout_manual_move_widget_EXAMPLE@
+-- @tparam widget widget The widget.
+-- @tparam table|function point A new point value.
+-- @see add_at
+function manual_layout:move_widget(widget, point)
+    local idx, l = self:index(widget, false)
+
+    if idx then
+        l:move(idx, point)
+    end
+end
+
+function manual_layout:get_children()
+    return self._private.widgets
+end
+
+function manual_layout:set_children(children)
+    self:reset()
+
+    self._private.widgets = children
```

Comment:
How about `:add(children)` here instead? Otherwise, the caller and this widget now both think they own the `children` table and are allowed to modify it.